ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m

```diff
@@ -26,6 +26,80 @@
 
 NSString* const AD_WPJ_LOG = @"ADNTLMHandler";
 
+@interface NTLMSheetHelper : NSObject
+{
+    dispatch_semaphore_t _dsem;
+    NSAlert*            _alert;
+    NSModalResponse     _response;
+    NSString*           _username;
+    NSString*           _password;
+}
+
+@property (readwrite, retain) NSString* username;
+@property (readwrite, retain) NSString* password;
+
+
+- (id)init;
+- (NSModalResponse)showNTLMSheet;
+
+@end
+
+@implementation NTLMSheetHelper
+
+@synthesize username = _username;
+@synthesize password = _password;
+
+- (id)init
+{
+    if (!(self = [super init]))
+        return nil;
+    
+    _dsem = dispatch_semaphore_create(0);
+
+    return self;
+}
+
+- (void)dealloc
+{
+    SAFE_ARC_RELEASE(_alert);
+    dispatch_release(_dsem);
+    SAFE_ARC_SUPER_DEALLOC();
+}
+
+- (void)internalShowNTLMSheet
+{
+    _alert = [NSAlert alertWithMessageText:NSLocalizedString(@"Enter your credentials", nil)
+                             defaultButton:NSLocalizedString(@"Login", nil)
+                           alternateButton:NSLocalizedString(@"Cancel", nil)
+                               otherButton:nil
+                 informativeTextWithFormat:@""];
+    
+    SAFE_ARC_RETAIN(_alert);
+    
+    ADCredentialCollectionController *view = [ADCredentialCollectionController new];
+    [view.usernameLabel setStringValue:NSLocalizedString(@"User Name", nil)];
+    [view.passwordLabel setStringValue:NSLocalizedString(@"Password", nil)];
+    [_alert setAccessoryView:view.customView];
+    
+    [_alert beginSheetModalForWindow:[NSApp keyWindow] completionHandler:^(NSModalResponse returnCode) {
+        _response = returnCode;
+        
+        _username = [view.usernameField stringValue];
+        _password = [view.passwordField stringValue];
```

Comment:
Add some SAFE_ARC_RETAINs to these fields or use [self setUsername:] & [self setPassword:] (or the dot syntax)